e2e/ansible/playbooks/hyperconverged/test-k8s-jenkins-upgrade/k8s-jenkins-upgrade.yml

```diff
@@ -0,0 +1,232 @@
+     - name: Copy job configuration files to jenkins-pod from master.
+       shell: |
+         source ~/.profile; kubectl cp template.xml  {{namespace}}/{{ jenkins_pod_name.stdout }}:/var/jenkins_home/
+         source ~/.profile; kubectl cp config_change.sh {{namespace}}/{{ jenkins_pod_name.stdout }}:/var/jenkins_home/
+       args:
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}" 
+   
+     - name: Run the config_change.sh file for required changes inside pod.
+       shell: > 
+          source ~/.profile; kubectl exec {{  jenkins_pod_name.stdout }} -n {{ namespace }} -- bash -c "sh /var/jenkins_home/config_change.sh"
+       args:
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+
+     - name: Retrieve Password of jenkins running pods.
+       shell: >
+          source ~/.profile; kubectl exec  {{ jenkins_pod_name.stdout }} -n {{ namespace }}
+          -- bash -c "cat /var/jenkins_home/secrets/initialAdminPassword"
+       args:
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+       register: jenkins_password
+  
+     - name: Restart the jenkins pod for running jenkins actions.
+       shell: >
+          source ~/.profile; kubectl exec {{ jenkins_pod_name.stdout }} -n {{ namespace }} 
+          -- java -jar /var/jenkins_home/jenkins-cli.jar -s http://localhost:8080 restart
+          --username admin --password {{ jenkins_password.stdout }}
+       args:
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+
+     - name: wait for 20 seconds.
+       wait_for:
+          timeout: 20
+ 
+     - name: Create and Build one sample job inside pod.
+       shell: |
+         source ~/.profile; kubectl exec {{ jenkins_pod_name.stdout }} -n {{ namespace }} -- bash -c "java -jar /var/jenkins_home/jenkins-cli.jar -s http://localhost:8080 create-job sample < /var/jenkins_home/template.xml"
+         source ~/.profile; kubectl exec {{ jenkins_pod_name.stdout }} -n {{ namespace  }} -- bash -c "java -jar /var/jenkins_home/jenkins-cli.jar -s http://localhost:8080 build sample" 
+       args:
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+    
+     - name: Running a build job
+       shell: >
+          source ~/.profile; kubectl exec {{ jenkins_pod_name.stdout }} -n {{ namespace }}
+          -- bash -c "java -jar /var/jenkins_home/jenkins-cli.jar -s http://localhost:8080 console sample"
+       args:
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}" 
+       register: First_Build
+  
+     - name: (2) Replace jenkins pod version 
+       lineinfile:
+         path: "{{ result_kube_home.stdout }}/{{ pod_yaml_alias }}"
+         regexp: "image: jenkins/jenkins:"
+         line: "     image: jenkins/jenkins:{{jenkinsnew}}"
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
```

Comment:
Can the regex_replace module be used (we anyways have the jenkins release versions as part of test vars